### PR TITLE
assign user to course

### PR DESCRIPTION
**Pull request by BuiDuySon**
assign user to course

### CODE REVIEW

File: app/controllers/courses_controller.rb [hunk 1]
```diff
@@ -1,36 +1,47 @@
...
+class CoursesController < ApplicationController
+  before_action :logged_in_user
+  before_action :set_course, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @courses = Course.all
+  end
+
+  def show
+    @enrollment_subjects = current_user.enrollment_subjects.where course: @course
```
Comment (chuanhtuan) [inline app/controllers/courses_controller.rb]: write in scope

File: app/controllers/suppervisor/courses_controller.rb [hunk 2]
```diff
@@ -7,15 +7,20 @@ def index
 
   def show
     @subjects = @course.subjects
+    @enrollment_subject = @course.enrollment_subjects.build(course_id: params[:id])
   end
 
   def new
     @course = Course.new
     @subjects = Subject.all
+    @users = User.where.not id: Enrollment.where(status: [0,1]).select(:user_id)
```
Comment (chuanhtuan) [inline app/controllers/suppervisor/courses_controller.rb]: scope

File: app/controllers/suppervisor/courses_controller.rb [hunk 3]
```diff
@@ -7,15 +7,20 @@ def index
...
   end
 
   def edit
     @subjects = Subject.all
+    @users = User.where.not id: Enrollment.
```
Comment (chuanhtuan) [inline app/controllers/suppervisor/courses_controller.rb]: scope

File: app/controllers/suppervisor/courses_controller.rb [hunk 4]
```diff
@@ -24,6 +29,7 @@ def create
       redirect_to suppervisor_courses_path
     else
       @subjects = Subject.all
+      @users = User.where.not id: Enrollment.where(status: [0,1]).select(:user_id)
```
Comment (chuanhtuan) [inline app/controllers/suppervisor/courses_controller.rb]: scope

File: app/controllers/suppervisor/courses_controller.rb [hunk 5]
```diff
@@ -32,6 +38,10 @@ def update
     if @course.update_attributes course_params
       redirect_to suppervisor_course_path(@course)
     else
+      @subjects = Subject.all
+      @users = User.where.not id: Enrollment.
```
Comment (chuanhtuan) [inline app/controllers/suppervisor/courses_controller.rb]: scope

File: app/controllers/courses_controller.rb [hunk 6]
```diff
@@ -1,36 +1,46 @@
...
+      flash[:danger] = "Please log in."
+      redirect_to login_url
+    end
+  end
+  
+  def set_course
+    @course = Course.find params[:id]
+  end
+
+  def course_params
```
Comment (chuanhtuan) [inline app/controllers/courses_controller.rb]: remove

File: app/controllers/suppervisor/enrollments_controller.rb [hunk 7]
```diff
@@ -0,0 +1,14 @@
+class Suppervisor::EnrollmentsController < ::BaseSuppervisorController
+  before_action :set_course, only: [:new, :create]
+
+  def new
+    @enrollment = @course.enrollments.new
+    @users = User.dont_join_any_courses + User.joining_with_course(@course) + User.finished_all_courses
```
Comment (dieunb) [inline app/controllers/suppervisor/enrollments_controller.rb]: `@user = User.all`

File: app/models/user.rb [hunk 8]
```diff
@@ -11,7 +13,11 @@ class User < ActiveRecord::Base
   validate file_size: {maximum: 1.megabytes.to_i}
 
   has_secure_password
+  
   scope :trainees, -> {where(role: 'trainee')}
```
Comment (dieunb) [inline app/models/user.rb]: `scope :trainees, ->{where(role: 'trainee')}`

File: app/models/user.rb [hunk 9]
```diff
@@ -11,7 +13,11 @@ class User < ActiveRecord::Base
...
+  scope :dont_join_any_courses, -> {includes(:enrollments).where(enrollments: {user: nil})}
```
Comment (dieunb) [inline app/models/user.rb]: `scope :dont_join_any_courses, ->{includes(:enrollments).where(enrollments: {user: nil})} `

File: app/models/user.rb [hunk 10]
```diff
@@ -11,7 +13,11 @@ class User < ActiveRecord::Base
   validate file_size: {maximum: 1.megabytes.to_i}
 
   has_secure_password
+  
   scope :trainees, -> {where(role: 'trainee')}
+  scope :dont_join_any_courses, -> {includes(:enrollments).where(enrollments: {user: nil})} 
+  scope :joining_with_course, ->(course) {includes(:enrollments).where(enrollments: {course: course, status: [0, 1]})}
+  scope :finished_all_courses, -> {includes(:enrollments).where(enrollments: {status: 2})}
```
Comment (dieunb) [inline app/models/user.rb]: `scope :finished_all_courses, ->{includes(:enrollments).where(enrollments: {status: 2})}`

**Pull request merged**